app/helpers/react_components/student/nudge.rb

```diff
@@ -24,8 +24,17 @@ def exercise_type
         solution.exercise.concept_exercise? ? "concept" : "practice"
       end
 
-      def iteration
-        SerializeIteration.(solution.iterations.order(id: :desc).first)
+      def request
+        {
+          endpoint: Exercism::Routes.api_solution_url(solution.uuid, sideload: [:iterations]),
+          options: {
+            initialData: {
+              iterations: solution.
+                iterations.
+                map { |iteration| SerializeIteration.(iteration) }
```

Comment:
Changing this from `iteration` to `iterations` is quite a lot of extra data. In the JS we only seem to use `iteration.last.status`. Do we need to pass all this other data here or have you only done that because we have a matching API endpoint? If so, I'd rather we added an extra endpoint that just gets the latest iteration's status, rather than having to pass so much data round.

Reply:
Yes. I just tweaked it because I wanted the endpoints to match. I will introduce a new endpoint and a new channel for the latest iteration.